featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -648,9 +637,7 @@ def _features_by_type(self, all_features, entity, max_depth,
             if (variable_type == variable_types.PandasTypes._all or
                     f.variable_type == variable_type or
                     any(issubclass(f.variable_type, vt) for vt in variable_type)):
-                if ((max_depth is None or self._get_depth(f) <= max_depth) and
-                        (self.max_hlevel is None or
-                         self._max_hlevel(f) <= self.max_hlevel)):
+                if max_depth is None or self._get_depth(f) <= max_depth:
```

Comment:
I think we can replace `self._get_depth(f)` with `f.get_depth(stop_at=self.seed_features)` and remove `self._get_depth`